api/src/plugins/cookie-update.ts

```diff
@@ -0,0 +1,37 @@
+import { FastifyPluginCallback } from 'fastify';
+
+import { unsign, type CookieSerializeOptions } from './cookies';
+
+type Options = { cookies: string[]; attributes: CookieSerializeOptions };
+
+/**
+ * Plugin that updates the attributes of cookies in the response, without
+ * changing the value.
+ *
+ * @param fastify The Fastify instance.
+ * @param options Options passed to the plugin via `fastify.register(plugin,
+ * options)`.
+ * @param options.cookies The names of the cookies to update.
+ * @param options.attributes The attributes to update the cookies with. NOTE:
+ * The attributes are merged with the default values given to \@fastify/cookie.
+ * @param done Callback to signal that the logic has completed.
+ */
+export const cookieUpdate: FastifyPluginCallback<Options> = (
+  fastify,
+  options,
+  done
+) => {
+  fastify.addHook('onSend', (request, reply, payload, next) => {
+    for (const cookie of options.cookies) {
+      const oldCookie = request.cookies[cookie];
+      if (!oldCookie) continue;
+
+      const unsigned = unsign(oldCookie);
+      const raw = unsigned.valid ? (unsigned.value as string) : oldCookie;
```

Comment:
Can we not type the unsign function better?

Something like `{ valid: false, value: null } | { valid: true, value: string }`?

Reply:
Yeah, we could do that, but I went down the rabbit hole a little and created a PR to fastify cookie: https://github.com/fastify/fastify-cookie/pull/292

If they're happy with that, we'll get an eslint error when we update and can just drop `as string` at that point.